### PR TITLE
Add benchmarking script, update API_DESIGN.md to reflect the results.

### PR DESCRIPTION
**Pull request by LukeWood**

### CODE REVIEW

File: .github/API_DESIGN.md [hunk 1]
```diff
@@ -4,15 +4,33 @@ In general, KerasCV abides to the  [API design guidelines of Keras](https://gith
...
 `segm` are often used.  In KerasCV, we will *not* be using these abbreviations.  This is done
 to ensure full consistency in our naming convention.  While the team is fond of the abbreviation
 `bbox`, we are loss fond of `segm`.  In order to ensure full consistency, we have decided to
 use the full names for label types in our code base.
 
-## Preprocessing Layers
-### Color Based Preprocessing Layers
+# Preprocessing Layers
+## BaseImageAugmentationLayer
+When implementing preprocessing, we encourage users to subclass the `tf.keras.__internal__.layers.BaseImageAugmentationLayer`.  This layer provides a common `call` method, auto vectorization, and more.
```
Comment (fchollet) [inline .github/API_DESIGN.md:16]: Nit: use `()` after method/function names, and keep lines under ~80 char to help with version control & reviews

File: benchmarks/vectorization_strategy_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+plt.legend()
+plt.show()
+
+"""
+# Sanity check
+all of these should have comparable outputs
+"""
+
+images = []
+for aug in [VectorizedRandomCutout, VMapRandomCutout, MapFnRandomCutout]:
```
Comment (fchollet) [inline benchmarks/vectorization_strategy_benchmark.py:680]: Would it be useful to also have numbers with XLA compilation?

**Review by fchollet — APPROVED** [review]

File: .github/API_DESIGN.md (hunk 1 above)
Comment (LukeWood) [reply .github/API_DESIGN.md]: Thanks - done.

File: benchmarks/vectorization_strategy_benchmark.py (hunk 2 above)
Comment (LukeWood) [reply benchmarks/vectorization_strategy_benchmark.py:680]: https://github.com/keras-team/keras-cv/issues/165 lets do in a follow up.

Comment (LukeWood) [reply benchmarks/vectorization_strategy_benchmark.py:680]: (but I do think so)

File: benchmarks/vectorization_strategy_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,700 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+# Setup/utils
+"""
+import time
+
+import matplotlib.pyplot as plt
```
Comment (bhack) [inline benchmarks/vectorization_strategy_benchmark.py:17]: Is the benchmark a test or an example?

https://github.com/keras-team/keras-cv/blob/master/setup.py#L30

File: .github/API_DESIGN.md [hunk 4]
```diff
@@ -1,28 +1,59 @@
...
+- `augment_bounding_box()` allows updates to bounding boxes to be made
+
+When a canonical layer subclassing BaseImageAugmentationLayer is available, a 
+link to it will be added here.
+
+## Vectorization
+`BaseImageAugmentationLayer` requires you to implement augmentations in an 
+image-wise basis instead of using a vectorized approach.  This design choice 
+was based made on the  results found in the 
+[vectorization\_strategy\_benchmark.py](../benchmarks/vectorization_strategy_benchmark.py)
```
Comment (bhack) [inline .github/API_DESIGN.md:35]: How we are going to run this from command-line?

File: benchmarks/vectorization_strategy_benchmark.py [hunk 5]
```diff
@@ -0,0 +1,700 @@
...
+    images.append(layer(x_train[:3]))
+images = [y for x in images for y in x]
+
+
+plt.figure(figsize=(8, 8))
+for i in range(9):
+    plt.subplot(3, 3, i + 1)
+    plt.imshow(images[i].numpy().astype("uint8"))
+    plt.axis("off")
+plt.show()
```
Comment (bhack) [inline benchmarks/vectorization_strategy_benchmark.py:691]: This is not going to work well in containers like in our official [keras dev container](https://github.com/keras-team/keras/tree/master/.devcontainer). Isn't better to default plot on file?

File: benchmarks/vectorization_strategy_benchmark.py (hunk 2 above)
Comment (bhack) [reply benchmarks/vectorization_strategy_benchmark.py:680]: In this case for Vectorized and VMap:

immagine

Comment (bhack) [reply benchmarks/vectorization_strategy_benchmark.py:680]: We also need to consider the general discussion we had at https://github.com/keras-team/keras-cv/pull/146#issuecomment-1049352918

As this one is a special case that is running fine with `vectorized_map/fallback_to_while_loop=False`.

Also,  this simple case doesn't have any specific not JITable ops like e.g. `ImageProjectiveTransformV3` (E.g. for rotation) or  `tf.bincount` (see https://github.com/keras-team/keras-cv/issues/141#issuecomment-1047817015)

**Pull request merged**